### ISSUE TITLE
Feature Request: Improved recording options and interface + task tray icon

### ISSUE THREAD

**Issue by vertigo220** [ShareX/ShareX#5749]
I don't think the recording icon (red dot) is sufficient for seeing when it's recording or for controlling the recording. When I first tested it, I couldn't even tell it was recording, because I was expecting some sort of pop-up like FastStone Capture has, which it turns out it has, but it doesn't show when recording the taskbar area, which makes sense but it just confused me. The real problem is that even though it has the separate tray icon, since I keep ShareX's icon in the overflow pop-up, the recording icon went there as well. And every time I want to do a recording, I have to move it out so I can see the recording icon. If possible, the two should be completely separate, so one can be hidden and the other showing.

Another thing I'd like to see, if possible, is the ability to pause and resume a recording instead of just stopping or aborting. I'd also like an option of what action to take when left-clicking the icon, since having it stop the recording could result in accidentally clicking on it and ending the recording prematurely. Having it show the current length and size of the recording, either in a tooltip or in the right-click menu, would be nice as well.

Another suggestion: when hovering over a window, when it auto-selects the window/region to record, allow middle-clicking to toggle the inclusion of the taskbar. I realize the region can be manually selected by dragging in order to include it, but automating it that way would be nice.

Also, it would be useful if it would provide an estimated MB/s and/or per minute for the selected options, both when encoding and when using the lossless mode. I realize it couldn't be exact, but just a rough approximation would be very helpful.

Finally, it shouldn't show (or allow to select) codecs that aren't available. For example, on my AMD system it allows me to choose QuickSync and NVENC. And AMD VCE should be added.

And, of course, thanks! I haven't been using ShareX long, but it's incredibly full-featured and seems to be pretty well-designed from what I've seen of it.

**Comment by tony** [comment]
Aside: Have you tried [ScreenToGif](https://www.screentogif.com) ([screenshots](https://www.screentogif.com/screenshots))? It's free and for Windows.

I use ShareX for screenshots in various ways, but for control of recording, ScreenToGif has it all, including a video editor and export features.

**Comment by yashpalgoyal1304** [comment]
- parahs - bad for readability. consider bulleting these. 
- also, u havent searched for duplicates either. for example:
   - abillity to pause
   - indication for start of recording

**Comment by vertigo220** [comment]
My first point isn't really a duplication of your issue. The only way it's related and in any sense a duplication is that both are related to knowing when recording has started, but yours is asking for a sound, whereas mine is simply an explanation of the problem followed by the conclusion that the separation of the icons, which you yourself said is new, would be a good step toward improving it.

As for paragraphs, most are just a couple lines, and I fail to see how bulleting them would be any different, as it would still be the same amount of content and in the same form, only with bullets. It's not exactly a wall of text, and not at all difficult to read. You may find it a bit much (though I hope not, because it's only about a minute's worth of reading), but it's better to include info for context and to help get the message across than short bullet points, which may not do as well as explaining, and can also come across as blunt.

I did miss the pause feature request when looking through issues, and as I suspected, it looks like it's not possible, which is unfortunate, but, again, expected. I've posted a thought on how to accomplish it in that issue, so we'll see what Jaex thinks of my proposal.

As for the codecs, if in fact ShareX is simply being fed that info by ffmpeg, then yes, that would be something to be handled there. It seems strange, though, that ffmpeg wouldn't have AMD VCE, and I would think ShareX could tell if a computer has an Intel vs AMD CPU and an Nvidia GPU, but I could of course be wrong.

**Comment by CubityFirst** [comment]
Literally most of your issues could be summerised in 1-2 sentences, people don't need to read a thesis literally every time you have a suggestion or have an issue.

**Comment by vertigo220** [comment]
TL;DR since reading is apparently hard: if it's more than 20 words and too much text for you, don't read it, my feelings won't be hurt, and we'll all be better off without your "input."

"Literally" most of my issues *were* 1-2 sentences, and the others were a whopping 3 and 5. Didn't realize a few hundred words is a "thesis." Sad that's a lot for you. And nobody made you read it, either. "People" don't need to read it, only the developer(s) and anyone interested in contributing code, and many want an actual explanation, both to provide justification for the request and to describe what's desired. And if they're too lazy to read ~15 sentences, they're probably not writing much code anyways.

Whereas your comment could have been *summarized* by saying nothing, as it's doing nothing but contributing to the very thing you're complaining about: adding more words.

**Comment by CubityFirst** [comment]
Reading isn't hard, you just have to consider that when there's 1 developer on a project, they don't want to spend hours and hours reading essays about individual issues.

If a problem needs a long explanation, fucking go for it. More power to you.
